components/list/list-item-drag-mixin.js

```diff
@@ -0,0 +1,100 @@
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { html } from 'lit-element/lit-element.js';
+import { nothing } from 'lit-html';
+
+export const ListItemDragMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			draggable: { type: Boolean, reflect: true },
+			key: { type: String, reflect: true }
+		};
+	}
+
+	constructor() {
+		super();
+		this._itemDragId = getUniqueId();
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		if (!this.key) {
+			this.draggable = false;
+		}
+	}
+
+	_dispatchDragEvent(detail) {
+		this.dispatchEvent(new CustomEvent('d2l-list-item-position', {
+			detail: detail,
+			bubbles: true
+		}));
+		return true;
```

Comment:
Is this helper intended to return `true`? Usually we would not return anything, but every once in a couple of places we may return the result of `dispatchEvent` (bool) which indicates whether the default behavior is prevented (by one of the handlers calling `e.preventDefault()`.